core/src/main/java/org/apache/iceberg/hadoop/wrappedio/BindingUtils.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.hadoop.wrappedio;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.function.Supplier;
+import org.apache.iceberg.common.DynMethods;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utility methods to assist binding to Hadoop APIs through reflection. Source: {@code
+ * org.apache.parquet.hadoop.util.wrapped.io.BindingUtils}.
+ */
+final class BindingUtils {
+
+  private static final Logger LOG = LoggerFactory.getLogger(BindingUtils.class);
+
+  private BindingUtils() {}
+
+  /**
+   * Load a class by name.
+   *
+   * @param className classname
+   * @return the class or null if it could not be loaded.
+   */
+  static Class<?> loadClass(String className) {
```

Comment:
FYI, Iceberg already has reflection helpers that we prefer. Take a look at `DynMethods` and similar classes.